src/dashboard/apps/consent/models.py

```diff
@@ -57,11 +58,37 @@ class Meta:  # noqa: D106
     def __str__(self):  # noqa: D105
         return f"{self.delivery_point} - {self.updated_at}: {self.status}"
 
+    def clean(self):
+        """Custom validation logic.
+
+        Validates and restricts updates to the Consent object if its status is set
+        to `VALIDATED`. This ensures that validated consents cannot be modified
+        after their status are defined to `VALIDATED` (We prevent this update
+        for contractual reasons).
+
+        Raises:
+        ------
+        ValidationError
+            If the Consent object's status is `VALIDATED`.
+        """
+        if self.pk:
+            try:
+                current_instance = Consent.objects.get(id=self.pk)
+            except Consent.DoesNotExist:
+                return
+
+            if current_instance.status == VALIDATED:
+                raise ValidationError(
+                    message=_("Validated consent cannot be modified once defined.")
+                )
+
     def save(self, *args, **kwargs):
         """Saves with custom logic.
 
         If the consent status is `REVOKED`, `revoked_at` is updated to the current time.
         """
+        self.clean()
```

Comment:
May the following be sufficient?

```suggestion
        if self.pk and self.status == VALIDATED:
            raise ValidationError(
                message=_("Validated consent cannot be modified once defined.")
            )
```

Reply:
clean mehod allows to display a message in the Django admin, otherwise it displays an error page.